src/views/activity.html

```diff
@@ -159,8 +159,41 @@
     <script src="/public/javascript/accessible-autocomplete.min.js"></script>
 
     <script>
+        const riversSelect = document.querySelector('#river')
+        const rivers = []
+        for (let i = 1; i < riversSelect.length; i++) {
+            rivers.push(riversSelect.options[i].text)
+        }
+
+        function normalise (string) {
+            const bracketIndex = string.indexOf('(')
+            const stringWithoutBracketEnding = bracketIndex >= 0 ? string.slice(0, bracketIndex) : string
+            return stringWithoutBracketEnding.toLowerCase().trim()
+        }
+
+        function score (name, query) {
+            if (name.startsWith(query)) {
+                return -2
+            }
+            if (name.includes(query)) {
+                return -1
+            }
+            return 0
+        }
+
+        function getRiversSuggestions (query, populateResults) {
+            const filteredRivers = rivers.map(river => ({ name: river, score: score(normalise(river), normalise(query)) }))
+                .filter(river => river.score < 0)
+                .sort((a, b) => {
+                    return a.score - b.score
+                }).map(river => river.name)
+
+            populateResults(filteredRivers)
+        }
+
         accessibleAutocomplete.enhanceSelectElement({
-            selectElement: document.querySelector('#river')
+            selectElement: document.querySelector('#river'),
+            source: getRiversSuggestions
```

Comment:
Please could you move these functions into a file and reference that instead? Sat here, they just increase the page weight and won't cache